app/src/main/java/com/duckduckgo/app/browser/DownloadConfirmationFragment.kt

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.Toast
+import androidx.core.content.FileProvider.getUriForFile
+import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.DownloadFileData
+import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.UserDownloadAction
+import com.duckduckgo.app.global.view.gone
+import com.duckduckgo.app.global.view.leftDrawable
+import com.duckduckgo.app.global.view.show
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.android.synthetic.main.download_confirmation.view.*
+import timber.log.Timber
+
+class DownloadConfirmationFragment(
+    private val downloadFileData: DownloadFileData,
+    private val userDownloadAction: UserDownloadAction
+) : BottomSheetDialogFragment() {
+
+    @SuppressLint("RestrictedApi")
+    override fun setupDialog(dialog: Dialog, style: Int) {
+        super.setupDialog(dialog, style)
+        val view = LayoutInflater.from(context).inflate(R.layout.download_confirmation, null)
+        dialog.setContentView(view)
+        setupViews(view)
+    }
+
+    private fun setupViews(view: View) {
+        view.download_message.text =
+            getString(getMessageResource(), downloadFileData.file.name)
+        view.open_with.setOnClickListener {
+            openFile()
+            dismiss()
+        }
+        view.replace.setOnClickListener {
+            userDownloadAction.acceptAndReplace()
+            dismiss()
+        }
+        view.continue_download.setOnClickListener {
+            userDownloadAction.accept()
+            dismiss()
+        }
+        view.cancel.setOnClickListener {
+            userDownloadAction.cancel()
+            dismiss()
+        }
+
+        if (downloadFileData.alreadyDownloaded) {
+            view.open_with.show()
+            view.replace.show()
+            view.continue_download.text = getString(R.string.keep_both)
+            view.continue_download.leftDrawable(R.drawable.ic_keepboth_24dp)
+        } else {
+            view.open_with.gone()
+            view.replace.gone()
+            view.continue_download.text = getString(R.string.fireContinue)
+            view.continue_download.leftDrawable(R.drawable.ic_file_24dp)
+
+        }
+    }
+
+    private fun getMessageResource() =
+        if (downloadFileData.alreadyDownloaded) R.string.file_already_downloaded
+        else R.string.download_filename
+
+
+    private fun openFile() {
+        val uri = getUriForFile(
```

Comment:
Place this variable declaration onto a single line. We use 150 char line breaks for this project.